addons/popochiu/editor/main_dock/tab_audio/tab_audio.gd

```diff
@@ -0,0 +1,353 @@
+			
+				# Fix #59 : remove the AudioCue from the A singleton
+				PopochiuResources.remove_audio_autoload(group, audio_cue_name, resource_path)
+				return false
+		)
+		
+		if group_data.is_empty():
+			PopochiuResources.erase_data_value("audio", group)
+		else:
+			group_data.sort_custom(
+				func (a: String, b: String) -> bool:
+					return PopochiuUtils.sort_by_file_name(a, b)
+			)
+			PopochiuResources.set_data_value("audio", group, group_data)
+	
+	await get_tree().process_frame
+	
+	# Remove freed objects in _audio_files_in_group
+	var to_erase := []
+	for key: String in _audio_files_in_group:
+		if not _audio_files_in_group[key] or not is_instance_valid(_audio_files_in_group[key]):
+			to_erase.append(key)
+	
+	for key in to_erase:
+		_audio_files_in_group.erase(key)
+
+
+func _put_audio_cues_in_group() -> void:
+	# Put already loaded (in popochiu_data.cfg) PopochiuAudioCues into their corresponding
+	# PopochiuGroup
+	for key: int in _groups:
+		var group_dic: Dictionary = _groups[key]
+		var group_data: Array = PopochiuResources.get_data_value("audio", group_dic.array, [])
+		
+		if group_data.is_empty(): continue
+		
+		for resource_path: String in group_data:
+			var ac: AudioCue = load(resource_path)
+			
+			if (
+				ac.audio.resource_path in _audio_files_in_group
+				and is_instance_valid(_audio_files_in_group[ac.audio.resource_path])
+			):
+				# TODO: Check if the resource_path has changed
+				continue
+			
+			var ar := _create_audio_cue_row(ac)
+			ar.cue_group = group_dic.array
+			(group_dic.group as PopochiuGroup).add(ar, true)
+			
+			_audio_files_in_group[ac.audio.resource_path] = ar
+
+
+func _create_audio_cue_row(audio_cue: AudioCue) -> HBoxContainer:
+	var ar: HBoxContainer = POPOCHIU_AUDIO_ROW_SCENE.instantiate()
+	
+	ar.name = audio_cue.resource_name
+	ar.path = audio_cue.resource_path
+	ar.audio_cue = audio_cue
+	ar.audio_tab = self
+	ar.stream_player = _asp
+	
+	ar.deleted.connect(_audio_cue_deleted)
+	ar.clicked.connect(_on_row_clicked)
+	
+	return ar
+
+
+func _create_audio_file_row(file_path: String) -> void:
+	var ar: HBoxContainer = POPOCHIU_AUDIO_ROW_SCENE.instantiate()
+	
+	ar.name = file_path.get_file().get_basename()
+	ar.path = file_path
+	ar.file_name = file_path.get_file()
+	ar.audio_tab = self
+	ar.stream_player = _asp
+	
+	ar.group_selected.connect(_create_audio_cue.bind(file_path, ar))
+	
+	unassigned_group.add(ar, true)
+	_audio_files_to_assign[file_path] = ar
+
+
+func _read_directory(dir: EditorFileSystemDirectory) -> void:
+	if not dir: return
+	
+	if dir.get_subdir_count():
+		for d in dir.get_subdir_count():
+			# Look out sub directories
+			_read_directory(dir.get_subdir(d))
+
+		# Look out the directory files
+		_read_files(dir)
+	else:
+		_read_files(dir)
+
+
+func _read_files(dir: EditorFileSystemDirectory) -> void:
+	for idx in dir.get_file_count():
+		var file_name = dir.get_file(idx)
+		
+		if not file_name.get_extension() in ["ogg", "mp3", "wav", "opus"]:
```

Comment:
Even this array may be stored somewhere as the one I mentioned above, maybe?

Reply:
Yup.

Reply:
Done.